pkg/query-service/rules/manager.go

```diff
@@ -295,6 +339,11 @@ func (m *Manager) DeleteRule(id string) error {
 		return err
 	}
 
+	err = m.updateFeatureUsage(&rule.PostableRule, -1)
+	if err != nil {
+		zap.S().Errorf("error updating feature usage: %v", err)
```

Comment:
It's not clear when do you return `err` vs `nil` vs nothing when there is an error from the feature manager functions.

Reply:
Updating feature usage is not critical enough to throw error. So just logging the error.

Reply:
> Updating feature usage is not critical enough to throw error

Isn't it the whole point of the feature gate to throw errors when, instead of just logging? that `GetFeature` could return `basemodel.ErrFeatureUnavailable` but we want to skip it silently?

Reply:
We are checking feature usuage initially in all the functions and returning error if usage limit exceeds.

Reply:
I follow that, but that check is only useful if the updates are successful. If I can find a way to fail the counter increments, I can exceed the limits and still keep the `checkFeatureUsage` happy.

Reply:
It is a very unlikely scenario.
We can execute `checkFeatureUsage` before create rule and add transactions over `checkFeatureUsage` and rollback if create rule has failed.

Reply:
We can do this later as it doesn’t affect functionality. But an added security measure to prevent misuse.